lib/tmuxinator/cli.rb

```diff
@@ -69,25 +71,86 @@ def completions(arg)
                           aliases: ["-l"],
                           desc: "Create local project file at ./.tmuxinator.yml"
 
-    def new(name)
-      project_file = find_project_file(name, options[:local])
-      Kernel.system("$EDITOR #{project_file}") || doctor
+    def new(name, session = nil)
+      if session
+        new_project_with_session(name, session)
+      else
+        new_project(name)
+      end
     end
 
     no_commands do
+      def new_project(name)
+        project_file = find_project_file(name, options[:local])
+        Kernel.system("$EDITOR #{project_file}") || doctor
+      end
+
+      def new_project_with_session(name, session)
```

Comment:
Just to be clear, I think this PR is well thought-out and makes for an excellent addition to tmuxinator's functionality.  I just think that breaking this down into more manageable chunks goes a long way to keeping complexity to a minimum, which increases robustness and longevity of the tool.  👍